benchadapt/python/benchadapt/client.py

```diff
@@ -48,8 +48,17 @@ def get(self, path: str) -> dict:
 
     def post(self, path: str, json: dict) -> Optional[dict]:
         url = self.base_url + path
-        log.debug(f"POST {url} {dumps(json)}")
-        res = self.session.post(url=url, json=json, timeout=self.timeout_s)
+
+        # log.debug(f"POST {url} {dumps(json)}")
+        print(f"POST {url} {dumps(json)}")
+
+        try:
+            res = self.session.post(url=url, json=json, timeout=self.timeout_s)
+        except requests.exceptions.ReadTimeout as e:
+            print(f"POST timed out: {e.response.content.decode()}. Retrying...")
+            print(f"POST {url} {dumps(json)}")
+            res = self.session.post(url=url, json=json, timeout=self.timeout_s)
```

Comment:
Can line 60 please be in another try except block so if it fails, we will continue posting other results in the loop?

We should still raise an error though if the any 2nd try failed but after the loop is done.

Reply:
K, moved all this to the posting loop so we can raise after